### DISCUSSION TITLE
Allow app to expose API

### DISCUSSION THREAD

**Discussion by taorepoara** [lenra-io/lenra_cli#101] [General]
Permitting the apps to expose an API to other apps in order to let them communicate.

For example, an agenda app could expose an API to allow the other apps to add an event.

This opens many questions:
- how to handle API update
- should we let the user allow the use of some endpoints of the API (and how to do it easely) or do we let the API creator manage roles and the user give roles access to the app API for a given time
- can the API be exposed to outside from Lenra apps ?

**Comment by jonas-martinez** [comment]
What do you mean by "How to handle API update" ? 

In my opinion, as long as the user has the permission to do a specific action on an app, he can use the corresponding endpoint to do the same thing. So this implies that the API creator must manages roles and access permissions.

Yes the API could be exposed to the outside, there is no real downside. We should still give the possibility to the developer (or user ?) to toggle this feature on or off.

**Reply by taorepoara** [reply]
The API can be updated when the app has new features or refactored. This could lead to breakable API changes that would need and update of the API clients.

The call of the API from outside Lenra would need an authentication of the caller, but it's need for other features so it's not a real problem.
